### DISCUSSION TITLE
how to: use a select with options from datasource, in a form using sql datasource

### DISCUSSION THREAD

**Discussion by dncpax** [Budibase/budibase#2664] [Help] [answered]
Hi there, again. I know it's Sunday so not expecting this to be answered today ;)

So while building a form to edit a single record, I'd like to have a select with options from a column in an "external" table. The form also is editing an "external" sql datasource.

What is the approach to this? I can't find a way, besides entering the options manually.
I can't use relationships bc it's an "external" query datasource.
I can't use a field in the form's datasource bc it's filtered to a single record so it can edited (so there's no way to get list of values - there's just a single record).

Can I use another datasource to fill the options?
Any other way?

Thanks.

**Comment by dncpax** [comment, marked as answer]
So, found a way, not sure if it's the canonical way, but here it is if someone is interested...
Turns out you can nest data providers and they are accessible to the form elements like the select.
So I've put a new data provider pointing to the table with the options at the top of the page hierarchy:

```
>Data provider with options from sql table
  >Data provider with filter id = {{page id parameter}}
    >Repeater
       >Form
          >Field Group
            >Select, with options from 1st data provider, shows all options
```
So this is working. Wondering if the team has other more appropriate ways?

Thanks.

**Reply by mjashanks** [reply]
Yep that's the way, except I'd usually put the dataprovider closer to the `Select` e.g.

```
  >Data provider with filter id = {{page id parameter}}
    >Repeater
       >Form
          >Field Group
            > Data provider with options from sql table
               >Select, with options from 1st data provider, shows all options
```

I don't think it really matters but reduces nesting in the tree. If you need that data in multiple places, you may want to do it as you describe.
